### ISSUE TITLE
Remove patch minor patch releases from buildpack

### ISSUE THREAD

**Issue by macsux** [cloudfoundry/dotnet-core-buildpack#518]
Buildpack currently carries a bunch of redundant patch releases for the same major/minor versions. There are very few practical reason to do this and it just bloats the size and slows down staging process.

The only way to peg to a specific patch version is by supplying `runtimeconfig.json` file which disables patch roll forward and this setting is not recommended by Microsoft. As a .NET architect with 20 years of experience, my take is:
a) I can't imagine a scenario where one would want to peg against a specific patch version
b) Most developers aren't even aware this is possible
c) Given rapid nature of patches being removed from buildpack, feature would be relatively useless
d) If a scenario comes up where they absolutely need to do this, they can always publish app with self-contained setting and bundle the runtime

The default is roll-forward minor when app is published.

As an example, [v2.3.39](https://github.com/cloudfoundry/dotnet-core-buildpack/releases/tag/v2.3.39) can be safely reduced down to this:

```
name	version	cf_stacks
bower	1.8.13	cflinuxfs3
dotnet-aspnetcore	3.1.22	cflinuxfs3
dotnet-aspnetcore	5.0.14	cflinuxfs3
dotnet-aspnetcore	6.0.2	cflinuxfs3
dotnet-runtime	3.1.22	cflinuxfs3
dotnet-runtime	5.0.14	cflinuxfs3
dotnet-runtime	6.0.2	cflinuxfs3
dotnet-sdk	3.1.416	cflinuxfs3
dotnet-sdk	5.0.405	cflinuxfs3
dotnet-sdk	6.0.102	cflinuxfs3
libgdiplus	6.1	cflinuxfs3
libunwind	1.6.2	cflinuxfs3
node	16.14.0	cflinuxfs3

```

**Comment by brayanhenao** [comment]
Thank you for this suggestion, certainly, the points you propose are quite accurate. I will discuss this with the team, we will keep you informed of any changes, thanks!

**Comment by brayanhenao** [comment]
@macsux The proposed changes were implemented and are available in version [v2.3.40](https://github.com/cloudfoundry/dotnet-core-buildpack/releases/tag/v2.3.40).
